tests/infer/test_valid_models.py

```diff
@@ -313,40 +314,94 @@ def guide():
     assert_ok(model, guide, trace_graph=trace_graph)
 
 
-def test_nested_iarange_iarange_warning():
+@pytest.mark.xfail(reason="https://github.com/uber/pyro/issues/370")
```

Comment:
Does this still xfail? It passes on my branch in #806 (with some changes) Ditto for all tests marked `xfail` below.